docs/design/ovs-pipeline.md

```diff
@@ -54,525 +54,606 @@
-additional dedicated tables to support the [Antrea-native policies](../antrea-network-policy.md)
-([AntreaPolicyEgressRuleTable] and [AntreaPolicyIngressRuleTable]).
-
-Consider the following Antrea ClusterNetworkPolicy (ACNP) in the Application tier as an
-example for the remainder of this document.
-
-```yaml
-apiVersion: crd.antrea.io/v1beta1
-kind: ClusterNetworkPolicy
-metadata:
-  name: cnp0
-spec:
-  priority: 10
-  tier: application # defaults to application tier if not specified
-  appliedTo:
-    - podSelector:
-        matchLabels:
-          app: server
-  ingress:
-    - action: Drop
-      from:
-        - podSelector:
-            matchLabels:
-              app: notClient
-      ports:
-        - protocol: TCP
-          port: 80
-  egress:
-    - action: Allow
-      to:
-        - podSelector:
-            matchLabels:
-              app: dns
-      ports:
-        - protocol: UDP
-          port: 53
+# Dump a group.
+kubectl exec -n kube-system <ANTREA_AGENT_POD_NAME> -c antrea-ovs -- ovs-ofctl dump-groups <BRIDGE_NAME> table=<TABLE_NAME> <GROUP_ID> -O Openflow15 [--no-stats] [--names]
 ```
 
-This ACNP is applied to all Pods with the `app: server` label in all
-Namespaces. For these Pods, it drops TCP traffic on port 80 from all
-Pods which have the `app: notClient` label. In addition to the ingress rules,
-this policy also allows egress UDP traffic on port 53 to all Pods with the
-label `app: dns`. Similar to K8s NetworkPolicy, Antrea will only install OVS
-flows for this ACNP on Nodes for which some of the Pods are the target of the
-policy. Thus, we have scheduled three Pods (appServer, appDns, appNotClient)
-on the same Node and they have the following IP addresses:
+where `<TABLE_NAME>` is the name of table name in the pipeline and `<GROUP_ID>`
+is the ID of the group.
 
-- appServer: 10.10.1.6
-- appNotClient: 10.10.1.7
-- appDns: 10.10.1.8
+## Registers
 
-## Tables
+We use various 32-bit OVS registers to carry information throughout the pipeline.
+
+
+| Register      | Field Range | Field Name                | RegMark Value | RegMark Name                    | Description                                                                      |
+|---------------|-------------|---------------------------|---------------|---------------------------------|----------------------------------------------------------------------------------|
+| NXM_NX_REG0   | bits 0-3    | PktSourceField            | 0x1           | FromTunnelRegMark               | Packet source is tunnel port.                                                    |
+| NXM_NX_REG0   | bits 0-3    | PktSourceField            | 0x2           | FromGatewayRegMark              | Packet source is Antrea gateway port.                                            |
+| NXM_NX_REG0   | bits 0-3    | PktSourceField            | 0x3           | FromLocalRegMark                | Packet source is local Pod.                                                      |
```

Comment:
nit: maybe a personal preference, but I feel like it would help a bit with readability if we did not repeat identical values across successive rows, and instead left the cell blank (since merging is not supported). For examples:

| Register      | Field Range | Field Name                | RegMark Value |
|---------------|-------------|---------------------------|---------------|
| NXM_NX_REG0   | bits 0-3    | PktSourceField            | 0x1           |
|   |     |            | 0x2           |
|   |     |            | 0x3           |
|   |     |            | 0x4           |
|   |     |            | 0x5           |
|   |     |            | 0x6           |
|   | bits 4-7    | PktDestinationField     | 0x1           |
|   |     |     | 0x1           |

Reply:
Much better!